### PR TITLE
[Utility] Add unit life bar

### PR DESCRIPTION
**Pull request by unprintable123**
添加了生命条并与原来的units兼容且同生命周期。
可以修改length，offset，color的设置。
虚继承大概只能自己再开一个Lifebar了然后hide原来的了。。。

### CODE REVIEW

**Comment by unprintable123** [comment]
添加了血量渐变的效果

File: src/battle_game/core/unit.h [hunk 1]
```diff
@@ -42,6 +45,13 @@ class Unit : public Object {
     health_ = std::clamp(new_health, 0.0f, 1.0f);
   }
 
+  virtual void RenderLifeBar() {
+    if (!lifebar_) {
+      lifebar_ = new LifeBar(game_core_, id_);
```
Comment (LazyJazz) [inline src/battle_game/core/unit.h:50]: We do not suggest using “new” in a formal project, could be unsafe.
Use smart pointers instead.

**Comment by LazyJazz** [comment]
I think you can integrate this feature into Unit class. It's not necessary to make it an individual object.

**Comment by LazyJazz** [comment]
After all, you've done a great job, and it would be better if you upload some screenshots of your work.

**Comment by unprintable123** [comment]
把lifebar改进去了（但设置不敢放出来，太乱了）
image

File: src/battle_game/core/unit.h [hunk 2]
```diff
@@ -6,9 +6,22 @@ namespace battle_game {
...
+class LifeBar {
+ public:
+  LifeBar();
+  bool display_{true};
+  glm::vec2 offset_{};
+  float length_{2.4f};
+  glm::vec4 front_color_{};
+  glm::vec4 background_color_{};
+  float fadeout_health_;
+  glm::vec4 fadeout_color_{};
```
Comment (LazyJazz) [inline src/battle_game/core/unit.h:18]: I mean, you can just put them into unit class. Or there's something else you were worried about.

**Review by LazyJazz — CHANGES_REQUESTED** [review]

**Comment by unprintable123** [comment]
改好了，之后觉得这些设置太乱了不负责（）

**Pull request merged**